proxy/src/schema.rs

```diff
@@ -0,0 +1,83 @@
+use juniper::{FieldResult, RootNode};
+use std::sync::Arc;
+
+use crate::source::{Address, Local, Project, Source};
+
+pub type Schema = RootNode<'static, Query, Mutation>;
+
+pub fn create() -> Schema {
+    Schema::new(Query {}, Mutation {})
+}
+
+#[derive(Clone)]
+pub struct Context {
+    source: Arc<Local>,
+}
+
+impl Context {
+    pub fn new(source: Local) -> Self {
+        Self {
+            source: Arc::new(source),
+        }
+    }
+}
+
+impl juniper::Context for Context {}
+
+pub struct Query;
+
+#[juniper::object(Context = Context)]
+impl Query {
+    fn apiVersion() -> &str {
+        "1.0"
+    }
+
+    fn all_projects(ctx: &Context) -> FieldResult<Vec<Project>> {
+        Ok(ctx.source.get_all_projects())
+    }
+
+    fn get_project(ctx: &Context, address: Address) -> FieldResult<Option<Project>> {
+        Ok(ctx.source.get_project(address))
+    }
+}
+
+#[test]
+fn test_schema_all_projects() {
+    use juniper::Variables;
+
+    let ctx = Context::new(Local::new());
+    let (res, _errors) = juniper::execute(
+        "query { allProjects { name } }",
```

Comment:
Could we maybe rename this to just `projects`? I know I had this in the schema initially, but now after looking at other examples it seems like it would be more elegant if we just use the plural form for collection endpoints.